// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum CloudbuildWorkerPoolStateEnum {
  CloudbuildWorkerPoolStateEnumNO_VALUE_DO_NOT_USE = 0;
  CloudbuildWorkerPoolStateEnumSTATE_UNSPECIFIED = 1;
  CloudbuildWorkerPoolStateEnumPENDING = 2;
  CloudbuildWorkerPoolStateEnumAPPROVED = 3;
  CloudbuildWorkerPoolStateEnumREJECTED = 4;
  CloudbuildWorkerPoolStateEnumCANCELLED = 5;
}

enum CloudbuildWorkerPoolPrivatePoolV1ConfigNetworkConfigEgressOptionEnum {
  CloudbuildWorkerPoolPrivatePoolV1ConfigNetworkConfigEgressOptionEnumNO_VALUE_DO_NOT_USE = 0;
  CloudbuildWorkerPoolPrivatePoolV1ConfigNetworkConfigEgressOptionEnumEGRESS_OPTION_UNSPECIFIED = 1;
  CloudbuildWorkerPoolPrivatePoolV1ConfigNetworkConfigEgressOptionEnumNO_PUBLIC_EGRESS = 2;
  CloudbuildWorkerPoolPrivatePoolV1ConfigNetworkConfigEgressOptionEnumPUBLIC_EGRESS = 3;
}

message CloudbuildWorkerPool {
  string name = 1;
  string display_name = 2;
  string uid = 3;
  map<string, string> annotations = 4;
  string create_time = 5;
  string update_time = 6;
  string delete_time = 7;
  CloudbuildWorkerPoolStateEnum state = 8;
  CloudbuildWorkerPoolPrivatePoolV1Config private_pool_v1_config = 9;
  string etag = 10;
  CloudbuildWorkerPoolWorkerConfig worker_config = 11;
  CloudbuildWorkerPoolNetworkConfig network_config = 12;
  string project = 13;
  string location = 14;
}

message CloudbuildWorkerPoolPrivatePoolV1Config {
  CloudbuildWorkerPoolPrivatePoolV1ConfigWorkerConfig worker_config = 1;
  CloudbuildWorkerPoolPrivatePoolV1ConfigNetworkConfig network_config = 2;
}

message CloudbuildWorkerPoolPrivatePoolV1ConfigWorkerConfig {
  string machine_type = 1;
  int64 disk_size_gb = 2;
}

message CloudbuildWorkerPoolPrivatePoolV1ConfigNetworkConfig {
  string peered_network = 1;
  string peered_network_ip_range = 2;
  CloudbuildWorkerPoolPrivatePoolV1ConfigNetworkConfigEgressOptionEnum egress_option = 3;
}

message CloudbuildWorkerPoolWorkerConfig {
  string machine_type = 1;
  int64 disk_size_gb = 2;
  bool no_external_ip = 3;
}

message CloudbuildWorkerPoolNetworkConfig {
  string peered_network = 1;
  string peered_network_ip_range = 2;
}

message ApplyCloudbuildWorkerPoolRequest {
  CloudbuildWorkerPool resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteCloudbuildWorkerPoolRequest {
  string service_account_file = 1;
  CloudbuildWorkerPool resource = 2;
}

message ListCloudbuildWorkerPoolRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListCloudbuildWorkerPoolResponse {
  repeated CloudbuildWorkerPool items = 1;
}

service CloudbuildWorkerPoolService {
  rpc ApplyCloudbuildWorkerPool(ApplyCloudbuildWorkerPoolRequest) returns (CloudbuildWorkerPool);
  rpc DeleteCloudbuildWorkerPool(DeleteCloudbuildWorkerPoolRequest) returns (google.protobuf.Empty);
  rpc ListCloudbuildWorkerPool(ListCloudbuildWorkerPoolRequest) returns (ListCloudbuildWorkerPoolResponse);
}
